backend/src/common/middlewares/auth.middleware.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  createParamDecorator,
+  ExecutionContext,
+  UnauthorizedException,
+} from "@nestjs/common";
+import { GqlExecutionContext } from "@nestjs/graphql";
+import { NextFunction, Request, Response } from "express";
+import { getMongoRepository } from "typeorm";
+import { User as UserEntity, Wallet } from "../../entities";
+
+// eslint-disable-next-line
+const Web3Token = require("web3-token");
+
+declare global {
+  // eslint-disable-next-line
+  namespace Express {
+    interface Request {
+      currentUserAddress: string | null;
+      user?: UserEntity;
+    }
+  }
+}
+
+export async function authMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) {
+  try {
+    const token =
+      req.headers["authorization"] || req.headers["Authorization"] || "";
+    const { address } = await Web3Token.verify(token);
+    req.currentUserAddress = address || null;
+
+    const wallet = await getMongoRepository(Wallet).findOne({ address });
+    const user = await getMongoRepository(UserEntity).findOne(wallet.userId);
```

Comment:
can we add the check here when wallet is not null @AtelyPham ?

Reply:
Ah a wallet always belongs to a user. So I skip that checking... Anyway, I will add the check for the code more robust :) Thanks anh <3